### DISCUSSION TITLE
Circular dependencies?

### DISCUSSION THREAD

**Discussion by midget214** [techno-tim/k3s-ansible#36] [Questions] [answered]
To preface this - I know very little about k3s in general - which is why I opted to learn k3s by using techno-tim's automated cluster creation playbook(s).

I've been trying to install traefik and rancher but it seems I am missing a key setup step (or I am making very wrong assumptions) which results in a cirular dependency appearing.

To install rancher we need a load balancer to point to the available master nodes.  From what I understand the idea is to use traefik for this. So lets install traefik.

According to techno-tim's documentation on traefik - it is recommended to use longhorn for the persisted storage volumes.  Ok lets install longhorn.

Following the documentation to install longhorn it installs successfully - but in order to get access to the UI - we need an ingress controller - and from what I understand - this is supposed to be traefik - which we are trying to install using longhorn as the persisted storage?

What am I missing?

**Comment by timothystewart6** [comment, marked as answer]
There's a new guide and even video on how to install traefik after getting k3s running!

https://docs.technotim.live/posts/kube-traefik-cert-manager-le/
